api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -33,11 +34,13 @@
 	private String name;
 	
 	private String description;
-	
+
 	private Set<Integer> memberIds;
```

Comment:
Earlier I commented that you need to remove this memberIds field and then update the setter and getter to call the new methods/logic you added i.e the setter should delagate to addMembership() while the getter should create the Set of the patient Ids of the memberships on the fly.